airbyte-workers/src/main/java/io/airbyte/workers/WorkerApp.java

```diff
@@ -385,175 +383,296 @@ static Optional<ContainerOrchestratorConfig> getContainerOrchestratorConfig(fina
     }
   }
 
-  private static void launchWorkerApp(final Configs configs, final DSLContext configsDslContext, final DSLContext jobsDslContext) throws IOException {
-    MetricClientFactory.initialize(MetricEmittingApps.WORKER);
+  private static AirbyteApiClient getApiClient(final Configs configs) {
+    final var authHeader = configs.getAirbyteApiAuthHeaderName();
+
+    // control plane workers communicate with the Airbyte API within their internal network, so https
+    // isn't needed
+    final var scheme = configs.isControlPlaneWorker() ? "http" : "https";
+
+    LOGGER.debug("Creating Airbyte Config Api Client with Scheme: {}, Host: {}, Port: {}, Auth-Header: {}",
+        scheme, configs.getAirbyteApiHost(), configs.getAirbyteApiPort(), authHeader);
+
+    final AirbyteApiClient airbyteApiClient = new AirbyteApiClient(
+        new io.airbyte.api.client.invoker.generated.ApiClient()
+            .setScheme(scheme)
+            .setHost(configs.getAirbyteApiHost())
+            .setPort(configs.getAirbyteApiPort())
+            .setBasePath("/api")
+            .setRequestInterceptor(builder -> {
+              builder.setHeader("User-Agent", "WorkerApp");
+              if (!authHeader.isBlank()) {
+                builder.setHeader(authHeader, generateAuthToken());
+              }
+            }));
+    return airbyteApiClient;
+  }
+
+  /**
+   * Generate an auth token based on configs. This is called by the Api Client's requestInterceptor
+   * for each request.
+   *
+   * For Data Plane workers, generate a signed JWT as described here:
+   * https://cloud.google.com/endpoints/docs/openapi/service-account-authentication
+   *
+   * Otherwise, use the AIRBYTE_API_AUTH_HEADER_VALUE from EnvConfigs.
+   */
+  private static String generateAuthToken() {
+    if (configs.isControlPlaneWorker()) {
+      // control plane workers communicate with the Airbyte API within their internal network, so a signed
+      // JWT isn't needed
+      return configs.getAirbyteApiAuthHeaderValue();
+    } else if (configs.isDataPlaneWorker()) {
+      try {
+        final Date now = new Date();
+        final Date expTime = new Date(System.currentTimeMillis() + TimeUnit.MINUTES.toMillis(JWT_TTL_MINUTES));
+        final String saEmail = configs.getDataPlaneServiceAccountEmail();
+        // Build the JWT payload
+        final JWTCreator.Builder token = JWT.create()
```

Comment:
would it make sense to move this jwt stuff into its own module? mainly i don't think we want to have JWT declared as a dependency in a bunch of different places. if we have relatively high confidence that we only need it here then we can keep it. even then i might move it to its own class so it could be tested separately. `WorkerApp` be busy.

Reply:
I think we can be reasonably confident about not needing this JWT token logic elsewhere, as this is a pretty specific strategy for authenticating from a Data Plane worker to a GCP-specific Control Plane. 

I think it would be a good move to move this to a separate class, one that instantiates an API Client for the Worker with the auth token logic built in. This class could eventually be injected via Micronaut to serve Cloud's purposes without leaking that logic into OSS code, and should also make it simpler to provide a WorkerScopedApiClient that can only hit the required sub-APIs as a future improvement.

Reply:
> I think it would be a good move to move this to a separate class, one that instantiates an API Client for the Worker with the auth token logic built in.

+ 1 sgtm.

Reply:
+1 on not needing this elsewhere. keeping this next to the api clients make sense.